### PR TITLE
Feature loanresult

### LINKED ISSUE #18: Define application result-messages
The format in which the bank replies needs to be defined.
CloudEvents stringyness in number-types should help with the issue of weird binary rounding of APR. The event should at least contain:
- BrokerReference
- Outcome (accept, decline, manual_processing)
- Effective Interest Rate
- Nominal Interest Rate
- Offered Credit
- EstablishingFee
- TermFee
- NumberOfYears

### PR DESCRIPTION
**Pull request by zinob**
this closes #18

### CODE REVIEW

File: spec.md [hunk 1]
```diff
@@ -178,7 +178,7 @@ Application
 | Name            | C.   | Type           | V. | Remark                              |
 |-----------------|------|----------------|----|-------------------------------------|
 | loanAmount      | 1    | number         | v0 | Total N/o SEK applied for           |
-| termYears       | 1    | number         | v0 | N/o desired years term for loan     |
+| termMonths      | 1    | number         | v0 | N/o desired years term for loan     |
```
Comment (williamhogman) [inline spec.md]: 👍

File: spec.md [hunk 2]
```diff
@@ -293,3 +293,74 @@ MaritalStatus
...
+Offer
+
+| Name                  | C.   | Type               | V. | Remark                                                                         |
+|-----------------------|------|--------------------|----|--------------------------------------------------------------------------------|
+| effectiveInterestRate | 1    | string             | v0 | A number formatted as a string to reduce risk of rounding error                |
+| nominalInterestRate   | 1    | string             | v0 | A number formatted as a string to reduce risk of rounding error                |
+| minOfferedCredit      | 1    | number             | v0 | The smallest amount that can be offered at the current nominal interest rate * |
+| offeredCredit         | 1    | number             | v0 | Offered amount, this is the value that the APR is based on.                    |
+| maxOfferedCredit      | 1    | number             | v0 | The largest amount that can be offered at the current nominal interest rate *  |
+| establishingFee       | 1    | number             | v0 | The initial payment for establishment of the loan                              |
```
Comment (williamhogman) [inline spec.md]: arrangementFee

File: spec.md [hunk 3]
```diff
@@ -293,3 +293,74 @@ MaritalStatus
...
+
+Offer
+
+| Name                  | C.   | Type               | V. | Remark                                                                         |
+|-----------------------|------|--------------------|----|--------------------------------------------------------------------------------|
+| effectiveInterestRate | 1    | string             | v0 | A number formatted as a string to reduce risk of rounding error                |
+| nominalInterestRate   | 1    | string             | v0 | A number formatted as a string to reduce risk of rounding error                |
+| minOfferedCredit      | 1    | number             | v0 | The smallest amount that can be offered at the current nominal interest rate * |
+| offeredCredit         | 1    | number             | v0 | Offered amount, this is the value that the APR is based on.                    |
+| maxOfferedCredit      | 1    | number             | v0 | The largest amount that can be offered at the current nominal interest rate *  |
```
Comment (williamhogman) [inline spec.md]: Should be optional?

File: spec.md [hunk 4]
```diff
@@ -293,3 +293,74 @@ MaritalStatus
...
+|-----------------|------|-------------|----|---------------------------------|
+| offer           | 1    | Application | v0 | Contains the offer for loan     |
+| broker          | 1    | reverse-dns | v0 | Domain-name of the broker       |
+| brokerReference | 1    | string      | v0 | Per-broker unique, not globally |
+
+Offer
+
+| Name                  | C.   | Type               | V. | Remark                                                                         |
+|-----------------------|------|--------------------|----|--------------------------------------------------------------------------------|
+| effectiveInterestRate | 1    | string             | v0 | A number formatted as a string to reduce risk of rounding error                |
```
Comment (williamhogman) [inline spec.md]: Define sub-type for numbers as strings?

File: spec.md [hunk 5]
```diff
@@ -293,3 +293,74 @@ MaritalStatus
...
+| nominalInterestRate   | 1    | string             | v0 | A number formatted as a string to reduce risk of rounding error                |
+| minOfferedCredit      | 1    | number             | v0 | The smallest amount that can be offered at the current nominal interest rate * |
+| offeredCredit         | 1    | number             | v0 | Offered amount, this is the value that the APR is based on.                    |
+| maxOfferedCredit      | 1    | number             | v0 | The largest amount that can be offered at the current nominal interest rate *  |
+| establishingFee       | 1    | number             | v0 | The initial payment for establishment of the loan                              |
+| termFee               | 1    | number             | v0 | The invoice cost or other fixed fees to be paid along with amortisation and interest every month |
```
Comment (williamhogman) [inline spec.md]: remove the word invoice cost as this assumes that is for the invoice that might be avoidable.
remove `The invoice cost or other` the rest of the sentence is very good

File: spec.md [hunk 6]
```diff
@@ -293,3 +293,74 @@ MaritalStatus
...
+| numberOfMonths        | 1    | number             | v0 | The offered term of the loan expressed as months                               |
```
Comment (williamhogman) [inline spec.md]: rename so as to be symmetrical with application

File: spec.md [hunk 7]
```diff
@@ -293,3 +293,74 @@ MaritalStatus
...
+| loanInsuranceOffer    | 0..1 | LoanInsuranceOffer | v0 | If any optional insurance is offered along with the loan                       |
+
+ ** If there is no leeway in the offer the min amount and the max amount should be set to the offeredCredit value
```
Comment (williamhogman) [inline spec.md]: Ok good this is fine.

File: spec.md (hunk 3 above)
Comment (williamhogman) [reply spec.md]: saw note below thought you preferred optional. I have no preference

File: spec.md [hunk 8]
```diff
@@ -293,3 +293,74 @@ MaritalStatus
...
+## PrivateUnsecuredLoanRejection
+
+This event pertains denials of unsecured loans for Swedish customers.
+
+| Name            | C.   | Type        | V. | Remark                          |
+|-----------------|------|-------------|----|---------------------------------|
+| rejectionReason | 0..1 | Application | v0 | Contains the loan application   |
+| broker          | 1    | reverse-dns | v0 | Domain-name of the broker       |
+| brokerReference | 1    | string      | v0 | Per-broker unique, not globally |
+| Cleanex         | 1    |             | v0 |                                 |
```
Comment (williamhogman) [inline spec.md]: remove this :trollface:

File: spec.md [hunk 9]
```diff
@@ -293,3 +293,74 @@ MaritalStatus
...
+| broker          | 1    | reverse-dns | v0 | Domain-name of the broker       |
+| brokerReference | 1    | string      | v0 | Per-broker unique, not globally |
+
+DelayReason
+
+| String value      | Remark |
+|-------------------|--------|
+| ManualProcessing  |        |
+| Holiday           |        |
+| OperationalIssues |        |
```
Comment (williamhogman) [inline spec.md]: UPPERCASE

File: spec.md (hunk 6 above)
Comment (williamhogman) [reply spec.md]: termMonths

File: spec.md (hunk 8 above)
Comment (zinob) [reply spec.md]: WTF i thought i did...

File: spec.md (hunk 3 above)
Comment (zinob) [reply spec.md]: I thought you preferred mandatory, i have no preference

File: spec.md (hunk 4 above)
Comment (zinob) [reply spec.md]: Yeaah we should...

File: spec.md (hunk 6 above)
Comment (zinob) [reply spec.md]: I accidentally started to write the comment in the variable-name. Like a boss

**Pull request merged**